src_py/sysfont.py

```diff
@@ -148,12 +148,112 @@ def initsysfonts_darwin():
     # disc
     elif exists("/usr/X11R6/bin/fc-list"):
         fonts = initsysfonts_unix("/usr/X11R6/bin/fc-list")
+    elif exists("/usr/sbin/system_profiler"):
+        fonts = initsysfonts_macos("/usr/sbin/system_profiler")
     else:
         fonts = {}
 
     return fonts
 
 
+def _add_sys_font_inner(current_font, fonts):
+
+    bold = 'bold' in current_font['style']
+    italic = 'italic' in current_font['style']
+    oblique = 'oblique' in current_font['style']
+
+    _addfont(
+        _simplename(current_font['full name']), bold, italic or oblique, current_font['path'],
+        fonts)
+
+
+def _add_sys_font(current_font, multiple_fonts, fonts):
+
+    if len(multiple_fonts) > 0:
+        for font_item in multiple_fonts:
+            _add_sys_font_inner(font_item, fonts)
+
+    if len(current_font) > 0:
+        _add_sys_font_inner(current_font, fonts)
+
+
+# read the fonts using system_profiler on macOS
+def initsysfonts_macos(path="/usr/sbin/system_profiler"):
+    """use system_profiler get a list of fonts"""
+    fonts = {}
+
+    arguments = "SPFontsDataType | grep -i -e 'location' -e 'family' -e 'style' -e 'name'"
+
+    try:
+        flout, flerr = subprocess.Popen('%s : %s' % (path, arguments), shell=True,
+                                        stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                        close_fds=True).communicate()
+    except Exception:
+        return fonts
+
+    entries = toascii(flout)
+
+    parsing_font = False
+    current_font = {}
+
+    multiple_fonts = []
+
+    lines = entries.split('\n')
+    lines_len = len(lines);
+                
+    try:
+        for idx in range(0, lines_len):
```

Comment:
no need of variable `lines_len` and `idx` , just loop over `lines` : 

```python
for line in lines:
  key, value = line.strip().split(':', 1)
  ...
```
also , if you do the `lower()` here : 
```python
for line in lines:
  key, value = line.strip().lower().split(':', 1)
  ...
```
then, you can remove all the other calls to `lower()` which would more clean imho ...

Reply:
can't use ```lower()``` on the whole line because mac's  file system is case sensitive and this line can contain a path to the font file, i fixed it for ```key```